### PR TITLE
fix: proxy event local variable

### PR DESCRIPTION
**Pull request by akoserwal**
## Pre-Submission Checklist

- [ X] Opened an issue discussing these changes before opening the PR
- [ X] Ran the linter and tests via `make prepush`
- [x] Included comprehensive and convincing tests for changes

## Issues
#1348 

Closes:

* Issue #1348 
* Issue #
* Issue #

> Summarize the issues that discussed these changes

Since PROXY_EVENTS are shared between every instance of a restify server, and upgrade is added but never removed, all future instances of Server will have the upgrade event proxied.

I haven't added unit-test. As this is my first PR. I like to get initial review. before working on the unit test.

# Changes

> What does this PR do?
update PROXY_EVENTS to an instance variable.

### CODE REVIEW

**Comment by hekike** [comment]
Yep, looks good to me.

**Comment by akoserwal** [comment]
@hekike : done

File: lib/server.js [hunk 1]
```diff
@@ -187,10 +187,10 @@ function Server(options) {
 
     this.router.on('mount', this.emit.bind(this, 'mount'));
 
-    if (!options.handleUpgrades && PROXY_EVENTS.indexOf('upgrade') === -1) {
-        PROXY_EVENTS.push('upgrade');
+    if (!options.handleUpgrades && this.proxyEvents.includes === false) {
```
Comment (DonutEspresso) [inline lib/server.js]: Do you mean `this.proxyEvents.includes('upgrade') === false`? That said, I don't think node 4 supports the includes method. @hekike we may have to stick with indexOf here for now?

**Comment by DonutEspresso** [comment]
Hi @akoserwal - thanks for the contribution! Would you mind adding unit tests to round out the PR?

File: lib/server.js (hunk 1 above)
Comment (hekike) [reply lib/server.js]: You are right, apologize for extra work :/

**Comment by akoserwal** [comment]
@DonutEspresso: Please can you review again. I did `make check-style` locally before pushing.  it shows clean. Checks failed due it

**Comment by DonutEspresso** [comment]
@hekike did we setup the make file to run prettier? What's the best way to autofix that stuff?

**Comment by akoserwal** [comment]
please review #1661: @DonutEspresso  @hekike

**Pull request closed without merging**